pyaedt/modules/fields_calculator.py

```diff
@@ -0,0 +1,356 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import copy
+import os
+
+import pyaedt
+from pyaedt.generic.general_methods import generate_unique_project_name
+from pyaedt.generic.general_methods import open_file
+from pyaedt.generic.general_methods import pyaedt_function_handler
+from pyaedt.generic.general_methods import read_toml
+
+
+class FieldsCalculator:
+    def __init__(self, app):
+        self.expression_catalog = read_toml(os.path.join(pyaedt.__path__[0], "misc", "expression_catalog.toml"))
+        self.__app = app
+        self.design_type = app.design_type
+        self.ofieldsreporter = app.ofieldsreporter
+
+    @property
+    def expression_names(self):
+        """List of available expressions.
+
+        Returns
+        -------
+        list
+        """
+        return list(self.expression_catalog.keys())
+
+    @pyaedt_function_handler()
+    def add_expression(self, calculation, assignment, name=None):
+        """Add named expression.
+
+        Parameters
+        ----------
+        calculation :
+            Calculation type.
+        assignment : int or :class:`pyaedt.modeler.cad.object3d.Object3d` or
+         :class:`pyaedt.modeler.cad.FacePrimitive
+            Name of the object from which to add the named expression.
```

Comment:
```suggestion
            Name of the object to add the named expression from.
```
We use less formal writing here as this is how we speak and is easier to comprehend.